core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -391,15 +404,17 @@ public void respond(ActionListener<TransportResponse.Empty> listener) {
         }
     }
 
-    class ReplicaOperationTransportHandler implements TransportRequestHandler<ReplicaRequest> {
+    class ReplicaOperationTransportHandler implements TransportRequestHandler<RequestWithAllocationID<ReplicaRequest>> {
         @Override
-        public void messageReceived(final ReplicaRequest request, final TransportChannel channel) throws Exception {
+        public void messageReceived(final RequestWithAllocationID<ReplicaRequest> request, final TransportChannel channel)
+            throws Exception {
             throw new UnsupportedOperationException("the task parameter is required for this operation");
         }
 
         @Override
-        public void messageReceived(ReplicaRequest request, TransportChannel channel, Task task) throws Exception {
-            new AsyncReplicaAction(request, channel, (ReplicationTask) task).run();
+        public void messageReceived(RequestWithAllocationID<ReplicaRequest> request, TransportChannel channel, Task task)
+            throws Exception {
+            new AsyncReplicaAction(request.request, request.allocationId, channel, (ReplicationTask) task).run();
```

Comment:
Maybe the parameter `request` needs a better name or the field `request` on the `RequestWithAllocationID` needs a better name? It's `request.request` that is triggering this comment.